bindings/python/iota_sdk/common.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2024 IOTA Stiftung
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+from json import dumps, JSONEncoder
+from enum import Enum
+import humps
+
+
+def custom_encoder(func, *args, **kwargs):
+    """The routine of dump json string.
+    """
+    class MyEncoder(JSONEncoder):
+        """Custom encoder
+        """
+        # pylint: disable=too-many-return-statements
+
+        def default(self, o):
+            to_dict_method = getattr(o, "to_dict", None)
+            if callable(to_dict_method):
+                return o.to_dict()
+            if isinstance(o, str):
+                return o
+            if isinstance(o, Enum):
+                return o.__dict__
+            if isinstance(o, dict):
+                return o
+            if hasattr(o, "__dict__"):
+                obj_dict = o.__dict__
+                items_method = getattr(self, "items", None)
+                if callable(items_method):
+                    for k, v in obj_dict.items():
+                        obj_dict[k] = dumps(v, cls=MyEncoder)
+                        return obj_dict
```

Comment:
is `return_obj_dict` correctly indented? feels like the loop should run through and not just return in the first iteration :thinking:

Reply:
Hm yeah probably not needed at all because it's returned in the line below afterwards then, removed it